pkg/workspace/creds.go

```diff
@@ -30,54 +31,72 @@ import (
 // credentials or tests interacting with one another
 const PulumiCredentialsPathEnvVar = "PULUMI_CREDENTIALS_PATH"
 
-// GetAccessToken returns an access token underneath a given key.
-func GetAccessToken(key string) (string, error) {
+// GetAccount returns an account underneath a given key.
+func GetAccount(key string) (Account, error) {
 	creds, err := GetStoredCredentials()
 	if err != nil && !os.IsNotExist(err) {
-		return "", err
+		return Account{}, err
 	}
-	if creds.AccessTokens == nil {
-		return "", nil
+	if creds.Accounts == nil {
+		token, ok := creds.AccessTokens[key]
+		if !ok {
+			return Account{}, nil
+		}
+		return Account{AccessToken: token}, nil
 	}
-	return creds.AccessTokens[key], nil
+	return creds.Accounts[key], nil
```

Comment:
This seems a little dangerous in the "upgrade to newer CLI" case as well, since if `StoreAccount` is ever called, the `Accounts` map will be initialized. (Thus on line 40, `creds.Accounts != nil`.) But we never "back port" the `Accounts` map with values from the `AccessTokens` map.

... so is it intentional that a `(Account{}, nil)` would be returned?

Reply:
Nope. I'll update this code to handle that case.